httplib.h

```diff
@@ -7162,16 +7162,23 @@ Server::process_request(Stream &strm, const std::string &remote_addr,
 #endif
 #endif
 
+  auto req_line = line_reader.ptr();
+
+  // Extract URI from the request line
+  std::string method, uri, version;
+  std::istringstream iss(req_line);
+  iss >> method >> uri >> version;
```

Comment:
This is effectively what `parse_request_line()` does. Shouldn't the check be moved there?